File: 1566detectPattern.js
var containsPattern = function (arr, m, k) {
  let count = 0;
  for (var i = 0; i + m < arr.length; i++) {
    if (arr[i] == arr[i + m]) {
      count++;
      if (count == m * (k - 1)) return true;
    } else count = 0;
  }
  return false;
};
console.log(containsPattern(((arr = [1, 2, 3, 1, 2]), (m = 2), (k = 2))));

File: 1566detectPattern.py
class Solution(object):
    def containsPattern(self, arr, m, k):
        cnt = 0
        for i in range(len(arr)-m):
            if arr[i] == arr[i+m]:
                cnt = cnt+1
                if(cnt == m*(k-1)):
                    return True
            else:
                cnt = 0
        return False
